silk/node/wpantund_base.py

```diff
@@ -147,10 +147,16 @@ def form(self, network, role):
         role = getattr(wpan, "ROLES")[role]
         self.store_data(role, self.role_label)
 
-        command = "form %s -T %s" \
-                  % (network.name, role)
+        command = "form {} -T {}".format(network.name, role)
 
-        command += " -c %s" % network.channel
+        command += " -c {}".format(network.channel)
+
+        if xpanid and not panid:
+            command += " -x {}".format(xpanid)
+        elif panid and not xpanid:
+            command += " -p {}".format(hex(panid))
+        elif xpanid and panid:
+            command += " -x {} -p {}".format(xpanid, hex(panid))
 
```

Comment:
Suggest to simplify line 154 to 159 in def form() :
if xpanid:
    command += ....
if panid:
    command += ....

Reply:
fixed